ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -556,16 +590,20 @@ private void read(SelectionKey key, Transmission transmission) throws IOExceptio
         transmission.onReceiveComplete();
         transmission.clearReceive();
       }
+    } catch (IOException e) {
+      handleReadWriteIOException(e, key);
```

Comment:
Let's handle this from within the poll method since `close` modifies thread unsafe data structs